### PR TITLE
feat(invitation-world): #MX-77

### PR DESCRIPTION
**Pull request by c4role**
## Describe your changes
Feature to send an email with all informations to connect to a Minetest's world. This invitation is also used to fill the whitelist, and then only users who have received invitation can access to the world.

https://entconf.gdapublic.fr/display/MINE/Minetest+Webservice+API

## Checklist tests
Create or import a world. Send invitation to one or several users, and then, try to access to the world with the information (user, mdp, address, port). Also try to access to another login. (it should not work)

## Issue ticket number and link
[MEX-77] https://entsupport.gdapublic.fr/browse/MEX-77

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by JinDouang — APPROVED** [review]

File: src/main/java/fr/openent/minetest/controller/MinetestController.java [hunk 1]
```diff
@@ -120,6 +118,17 @@ public void putStatus(final HttpServerRequest request) {
                             .onFailure(err -> renderError(request))));
     }
 
+    @Put("/world/join")
+    @ApiDoc("Update status world")
+    @SecuredAction(value = "", type = ActionType.AUTHENTICATED)
+    public void joinWorld(final HttpServerRequest request) {
+        RequestUtils.bodyToJson(request, pathPrefix + Field.JOIN, body
+                -> UserUtils.getUserInfos(eb, request, user
+                -> worldService.join(user, body, request)
```
Comment (sjollois) [inline src/main/java/fr/openent/minetest/controller/MinetestController.java:127]: request -> translate
à refactoriser pour le modeliser et ne plus utiliser request dans la definition

File: src/main/resources/public/ts/directives/end-user-typing/end-user-typing.directive.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import {ng} from 'entcore';
+
```
Comment (sjollois) [inline src/main/resources/public/ts/directives/end-user-typing/end-user-typing.directive.ts:2]: Directive utilisée dans le template de la directive recipient-list
A remettre au propre

File: src/main/resources/public/ts/directives/recipient-list/recipient-list.directive.ts [hunk 3]
```diff
@@ -0,0 +1,200 @@
+import {ng, _, toasts} from "entcore";
+import {User, Users} from "../../models";
+import {minetestService} from "../../services";
+import {safeApply} from "../../utils/safe-apply.utils";
+import {RootsConst} from "../../core/constants/roots.const";
+
+
```
Comment (sjollois) [inline src/main/resources/public/ts/directives/recipient-list/recipient-list.directive.ts:7]: Directive utilisée dans le template de la directive invitation-world
A remettre au propre

File: src/main/resources/public/ts/models/user.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
+﻿import {idiom as lang, _, toasts} from "entcore";
+
+import { Mix, Selectable, Eventer } from "entcore-toolkit";
+import {minetestService} from "../services";
+import {AxiosResponse} from "axios";
+
```
Comment (sjollois) [inline src/main/resources/public/ts/models/user.ts:6]: Classes utilisées dans la directive recipient-list
A remettre au propre

**Review by sjollois — COMMENTED** [review]
Après Mis en production remettre au propre les directives et classes pour la maintenabilité du module

**Pull request merged**